lib/time/time.go

```diff
@@ -0,0 +1,454 @@
+		"sleep":         starlark.NewBuiltin("sleep", sleep),
+		"fromtimestamp": starlark.NewBuiltin("fromtimestamp", fromtimestamp),
+
+		"zero": Time{},
+
+		"nanosecond":  durInt(gotime.Nanosecond),
+		"microsecond": durInt(gotime.Microsecond),
+		"millisecond": durInt(gotime.Millisecond),
+		"second":      durInt(gotime.Second),
+		"minute":      durInt(gotime.Minute),
+		"hour":        durInt(gotime.Hour),
+	},
+}
+
+// LoadModule loads the time module.
+// It is concurrency-safe and idempotent.
+func LoadModule() (starlark.StringDict, error) {
+	return starlark.StringDict{
+		ModuleName: Module,
+	}, nil
+}
+
+// NowFunc is a function that generates the current time. Intentionally exported
+// so that it can be overridden
+var NowFunc = func() gotime.Time { return gotime.Now() }
+
+func duration(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var x starlark.String
+	if err := starlark.UnpackArgs("duration", args, kwargs, "x", &x); err != nil {
+		return nil, err
+	}
+
+	d, err := gotime.ParseDuration(string(x))
+	if err != nil {
+		return nil, err
+	}
+
+	return Duration(d), nil
+}
+
+func durInt(d gotime.Duration) starlark.Int {
+	return starlark.MakeInt64(int64(d))
+}
+
+func location(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var x starlark.String
+	if err := starlark.UnpackArgs("location", args, kwargs, "x", &x); err != nil {
+		return nil, err
+	}
+	loc, err := gotime.LoadLocation(string(x))
+	if err != nil {
+		return nil, err
+	}
+
+	return starlark.String(loc.String()), nil
+}
+
+func sleep(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var x starlark.Int
+	if err := starlark.UnpackArgs("sleep", args, kwargs, "x", &x); err != nil {
+		return starlark.None, err
+	}
+	dur, ok := x.Int64()
+	if !ok {
+		return starlark.None, fmt.Errorf("invalid sleep value")
+	}
+
+	gotime.Sleep(gotime.Duration(dur))
+	return starlark.None, nil
+}
+
+func time(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		x, location starlark.String
+		format      = starlark.String(gotime.RFC3339)
+	)
+	if err := starlark.UnpackArgs("time", args, kwargs, "x", &x, "format?", &format, "location", &location); err != nil {
+		return nil, err
+	}
+
+	if location == "" {
+		t, err := gotime.Parse(format.String(), x.String())
+		if err != nil {
+			return nil, err
+		}
+		return Time(t), nil
+	}
+
+	loc, err := gotime.LoadLocation(location.String())
+	if err != nil {
+		return nil, err
+	}
+	t, err := gotime.ParseInLocation(format.String(), x.String(), loc)
+	if err != nil {
+		return nil, err
+	}
+	return Time(t), nil
+}
+
+func fromtimestamp(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
```

Comment:
"timestamp "is insufficiently precise: it's a 32-bit number of seconds since 1970, aka a UNIX timestamp. 

from_unixtime ?